addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -23,44 +68,41 @@ options.registry.mailing_list_subscribe = options.Class.extend({
      *
      * @see this.selectClass for parameters
      */
-    select_mailing_list: function (previewMode, value) {
-        var self = this;
-        var def = wUtils.prompt({
-            'id': this.popup_template_id,
-            'window_title': this.popup_title,
-            'select': _t("Newsletter"),
-            'init': function (field, dialog) {
-                return rpc.query({
-                    model: 'mailing.list',
-                    method: 'name_search',
-                    args: ['', [['is_public', '=', true]]],
-                    context: self.options.recordInfo.context,
-                }).then(function (data) {
-                    $(dialog).find('.btn-primary').prop('disabled', !data.length);
-                    var list_id = self.$target.attr("data-list-id");
-                    $(dialog).on('show.bs.modal', function () {
-                        if (list_id !== "0"){
-                            $(dialog).find('select').val(list_id);
-                        };
-                    });
-                    return data;
-                });
-            },
-        });
-        def.then(function (result) {
-            self.$target.attr("data-list-id", result.val);
-        });
-        return def;
+    selectMailingList(previewMode, widgetValue, params) {
+        this.$target.attr("data-list-id", widgetValue);
     },
     /**
      * @override
      */
     onBuilt: function () {
-        var self = this;
         this._super();
-        this.select_mailing_list('click').guardedCatch(function () {
-            self.getParent()._onRemoveClick($.Event( "click" ));
-        });
+        const mailingListID = parseInt(this.$target.attr('data-list-id')) || this.defaultMailingID;
+        if (mailingListID) {
+            this.$target.attr("data-list-id", mailingListID);
+        } else {
+            this._showWarning();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async _renderCustomXML(uiFragment) {
+        await this._renderMailingListButtons(uiFragment);
+    },
+    /**
+     * @override
+     */
+    _computeWidgetState(methodName, params) {
+        switch (methodName) {
+            case 'selectMailingList':
+                return parseInt(this.$target.attr('data-list-id')) || this.defaultMailingID;
+        }
+        return this._super(...arguments);
```

Comment:
This can be removed and is automatically with select-data-attribute on the option